WordPress/gutenberg
# Site Logo: Apply width to logo container in editor

## What?

When the width property is set in the block markup for the Site Logo block but there is no image applied, the Site Logo placeholder will default to filling the available width. This may happen if a theme or pattern specifies a set width in the block markup, for example:

```

```

## Why?

Themes or patterns should be able to provide an ideal Site Logo width that is represented in the editor before an image is added. Fixes https://github.com/WordPress/gutenberg/issues/45788.

## How?

This PR applies the `width` property, if it exists, to the Site Logo placeholder in the editor.

~I also tried applying the width to the placeholder itself (on line 510), but this means the block container still fills 100% of the width when selected, as shown with the selected blue border. I thought it seemed like a better user experience for the entire block to be limited to the specified width.~

## Testing Instructions




(Taken from the above issue.)

1. Make a pattern with a site logo. (Or use the markup below)
2. Add an image to the block and change the size to whatever but not-default. (Or use the markup below)
3. Copy the markup and paste it to another site that has no site logo.
4. The width of the placeholder should match the specified width.

Sample markup:

```

<div class="wp-block-group alignfull" style="padding-top:2em;padding-bottom:2em">



</div>

```

## Screenshots or screencast 

Using the above markup:

| Before | After |
| ------ | ----- |
| <img width="1215" alt="image" src="https://user-images.githubusercontent.com/1645628/202261664-b128ae19-5fea-4917-9026-cf7562ecd846.png">| <img width="1215" alt="image" src="https://user-images.githubusercontent.com/1645628/202261546-0e365c4a-81db-489f-9608-9aee4d66f8fe.png"> |

cc @WordPress/block-themers
